### PR TITLE
test: add tests for most used admin transactions

### PR DESCRIPTION
**Pull request by kingpinXD**
# Description

Closes https://github.com/zeta-chain/node/issues/2448

# How Has This Been Tested?

- [ ] Tested CCTX in localnet
- [ ] Tested in development environment
- [ ] Go unit tests
- [ ] Go integration tests
- [ ] Tested via GitHub Actions

## Summary by CodeRabbit

- **New Features**
  - Implemented new critical admin transaction tests for enhanced blockchain system management.
  - Added `Inbound Tracker` query support for tracking Ethereum and Bitcoin transactions.

- **Improvements**
  - Updated admin-related test constants for better clarity and organization.
  - Introduced methods for block height management to improve testing capabilities.
  - Added a command for displaying inbound tracker details with improved error handling.

### CODE REVIEW

File: e2e/e2etests/e2etests.go [hunk 1]
```diff
@@ -533,6 +534,12 @@ var AllE2ETests = []runner.E2ETest{
 		[]runner.ArgDefinition{},
 		TestRateLimiter,
 	),
+	runner.NewE2ETest(
+		TestCriticalAdminTransactionsName,
+		"test critical admin transactions",
```
Comment (lumtis) [inline e2e/e2etests/e2etests.go:539]: Can we provide more details here? What is a critical transaction?

File: e2e/e2etests/e2etests.go [hunk 2]
```diff
@@ -533,6 +534,12 @@ var AllE2ETests = []runner.E2ETest{
...
+		[]runner.ArgDefinition{},
+		TestAdminTransactions,
```
Comment (lumtis) [inline e2e/e2etests/e2etests.go:541]: Function name should be consistent with test name, use critical or not in the name

**Comment by lumtis** [comment]
Seems like we need to run `make generate`

**Review by lumtis — APPROVED** [review]

File: e2e/e2etests/test_admin_transactions.go [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+// MsgUpdateTssAddress
+//
+//	However, the transactions other than `AddToInboundTracker` and `UpdateGasPriceIncreaseFlags` have already been used in other tests.
+func TestCriticalAdminTransactions(r *runner.E2ERunner, _ []string) {
+	TestAddToInboundTracker(r)
+	TestUpdateGasPriceIncreaseFlags(r)
+}
+
+func TestUpdateGasPriceIncreaseFlags(r *runner.E2ERunner) {
+	defaultFlags := observertypes.DefaultGasPriceIncreaseFlags
```
Comment (swift1337) [inline e2e/e2etests/test_admin_transactions.go:37]: Can you please add comments to this test case? Also you can use "arrange", "act", "assert" style. It really helps to grasp the context of the test

File: x/crosschain/client/cli/query_inbound_tracker.go [hunk 4]
```diff
@@ -11,6 +11,32 @@ import (
...
+	cmd := &cobra.Command{
+		Use:   "show-inbound-tracker [chainID] [txHash]",
+		Short: "shows an inbound tracker by chainID and txHash",
+		Args:  cobra.ExactArgs(2),
+		RunE: func(cmd *cobra.Command, args []string) (err error) {
+			clientCtx := client.GetClientContextFromCmd(cmd)
+			queryClient := types.NewQueryClient(clientCtx)
+			argChain, err := strconv.ParseInt(args[0], 10, 64)
+			if err != nil {
+				return err
```
Comment (swift1337) [inline x/crosschain/client/cli/query_inbound_tracker.go:24]:
```suggestion
				return errors.Wrap(err, "unable to parse chain id from %q", args[0])
```

Comment (swift1337) [reply x/crosschain/client/cli/query_inbound_tracker.go:24]: This also applies to other errors if this method

File: x/crosschain/keeper/grpc_query_inbound_tracker_test.go [hunk 5]
```diff
@@ -48,3 +48,36 @@ func TestKeeper_InboundTrackerAll(t *testing.T) {
...
+		chainID := chains.GoerliLocalnet.ChainId
+		k.SetInboundTracker(ctx, types.InboundTracker{
+			ChainId:  chainID,
+			TxHash:   hash,
+			CoinType: coin.CoinType_Gas,
+		})
+
+		res, err := k.InboundTracker(ctx, &types.QueryInboundTrackerRequest{
+			ChainId: chainID,
+			TxHash:  hash,
```
Comment (swift1337) [inline x/crosschain/keeper/grpc_query_inbound_tracker_test.go:65]: let's check that hash matches

**Review by swift1337 — APPROVED** [review]
LGTM. left some minor non-blocking comments.

File: x/crosschain/keeper/grpc_query_inbound_tracker_test.go (hunk 5 above)
Comment (kingpinXD) [reply x/crosschain/keeper/grpc_query_inbound_tracker_test.go:65]: That is part of the key; If the value is fetched, it should mean that the hash matches; is there an edge case you are referring to here?

**Pull request merged**